src/test/java/htsjdk/samtools/util/CigarElementUnitTest.java

```diff
@@ -12,4 +14,28 @@
     public void testNegativeLengthCheck(){
         final CigarElement element = new CigarElement(-1, CigarOperator.M);
     }
+
+
+    @DataProvider
+    public Object[][] elementsForEquals() {
+        final CigarElement mElement = new CigarElement(10, CigarOperator.M);
+        return new Object[][] {
+                // same object
+                {mElement, mElement, true},
+                // different equal objects
+                {mElement, new CigarElement(mElement.getLength(), mElement.getOperator()), true},
+                // different lengths
+                {mElement, new CigarElement(mElement.getLength() + 1, mElement.getOperator()), false},
+                // different operators
+                {mElement, new CigarElement(mElement.getLength(), CigarOperator.X), false},
+                // different class
+                {mElement, mElement.toString(), false}
+        };
+    }
+
+    @Test(dataProvider = "elementsForEquals")
+    public void testEqualsAndHashCode(final CigarElement element, final Object other, final boolean isEquals) {
+        Assert.assertEquals(element.equals(other), isEquals);
+        Assert.assertEquals(element.hashCode() == other.hashCode(), isEquals);
```

Comment:
I think this test isn't technically correct.  It's possible 
for `A != B` but `A.hashCode() == B.hashCode()`.   I think the correct assertion for the the hashCode part should be.

```
if( element.equals(other)) {
 Assert.assertEquals(element.hashCode(), other.hashCode());
}
```